### PR TITLE
Ruby: SimpleParameter should not be an Expr

### PR DESCRIPTION
**Pull request by aibaars**
PR #7394 makes all expressions "post-order" nodes in the control flow graph except for [`SimpleParameter`](https://github.com/github/codeql/pull/7394/files#diff-32ce71534428bfb48247fdc1f6ca40704b369d31bb38b73e2972cadd97367b1aR1) . I think parameters should not be instance of `Expr` , therefore, this pull requests removes the `Expr` class from the super class list of `SimpleParameter`.

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/ast/Erb.qll [hunk 1]
```diff
@@ -159,7 +159,7 @@ class ErbDirective extends TDirectiveNode, ErbAstNode {
    */
   Stmt getAChildStmt() {
     this.containsStmtStart(result) and
-    not this.containsStmtStart(result.getParent())
+    not this.containsStmtStart(parent*(result).getParent())
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/Erb.qll:162]: Why this unrelated change?

**Review by hvitved — COMMENTED** [review]
LGTM, but I don't understand the ERB change.

File: ruby/ql/lib/codeql/ruby/ast/Erb.qll (hunk 1 above)
Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/ast/Erb.qll:162]: It's not unrelated. One of the synthesized variable accesses has a SimpleParameter as parent and because SimpleParameter is not longer a Stmt( Expr extends Stmt) this negation suddenly held. The `parent*` change is to find the nearest parent that is instanceof `Stmt`.

**Review by alexrford — DISMISSED** [review]

File: ruby/ql/lib/codeql/ruby/ast/Erb.qll (hunk 1 above)
Comment (hvitved) [reply ruby/ql/lib/codeql/ruby/ast/Erb.qll:162]: What if we generalize `containsStmtStart(Stmt s)` to `containsNodeStart(AstNode n)` instead?

Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/ast/Erb.qll]: That should work too.

**Review by hvitved — APPROVED** [review]
